web/src/components/vis/ScreePlot.vue

```diff
@@ -156,68 +164,41 @@ export default {
           result[0] = i;
         }
       }
-
       return result;
-    },
-
-  },
-  watch: {
-    eigenvalues() {
-      this.update();
-    },
-    numComponents() {
-      this.update();
-    },
-    showCutoffs() {
-      this.update();
-    },
-  },
-  methods: {
-    update() {
-      const {
-        eigenvaluesInternal,
-        percents,
-        cumulativePercents,
-        fadeInDuration,
-        duration,
-        numComponentsInternal,
-        showCutoffs,
-        cutoffs,
-      } = this;
-
-      if (this.eigenvaluesInternal.length === 0) {
-        return;
-      }
-
-      const radius = 4;
-
-      if (!this.$refs.svg) {
+    });
+
+    function update() {
+      // Recalculate width/height
+      const bb = mainRef.value.getBoundingClientRect();
```

Comment:
Here, for example, in order to prove to myself this isn't null, I have to look at all the places that call update.